### PR TITLE
Add AMQP executor

### PR DESCRIPTION
**Pull request by chris1786**
Add AMQP executor

This is primarily designed to support recent versions of QPID, but should also be compatible with ActiveMQ, and other brokers which implement the AMQP 1.0 specification.

Signed-off-by: chris1786 <22242927+chris1786@users.noreply.github.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#637.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-venom-a#637.1 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

**Review by yesnault — COMMENTED** [review]
Thank you @chris1786  !

LGTM - Tests OK with last master.

**Review by yesnault — APPROVED** [review]

File: executors/amqp/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- Addr         (address of the amqp broker to connect to, in the format "amqp://<host>:<port>")
+- ClientType   (consumer or producer)
+
+# Consumer Parameters
+- SourceAddr   (source topic/queue to consume messages from)
+- MessageLimit (number of messages to read from the broker before returning result)
+
+# Producer Parameters
+- TargetAddr   (topic/queue to which messages should be published)
+- Messages     (array of message bodies to send to the broker)
```
Comment (fsamin) [inline executors/amqp/README.md:17]:
```suggestion
- addr         (address of the amqp broker to connect to, in the format "amqp://<host>:<port>")
- clientType   (consumer or producer)

# Consumer Parameters
- sourceAddr   (source topic/queue to consume messages from)
- messageLimit (number of messages to read from the broker before returning result)

# Producer Parameters
- targetAddr   (topic/queue to which messages should be published)
- messages     (array of message bodies to send to the broker)
```

**Review by fsamin — CHANGES_REQUESTED** [review]
I suggest to write the right yaml properties in the README.

Otherwise LGTM, thanks you

File: executors/amqp/README.md (hunk 1 above)
Comment (chris1786) [reply executors/amqp/README.md]: Thanks @fsamin - not sure how I missed this!

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Pull request merged**